pkg/apis/kops/validation/validation.go

```diff
@@ -430,25 +430,24 @@ func validateNetworkingFlannel(v *kops.FlannelNetworkingSpec, fldPath *field.Pat
 func validateNetworkingCanal(v *kops.CanalNetworkingSpec, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 
-	action := v.DefaultEndpointToHostAction
-	switch action {
-	case "", "ACCEPT", "DROP", "RETURN":
-	default:
-		allErrs = append(allErrs, field.NotSupported(fldPath.Child("defaultEndpointToHostAction"), action, []string{"ACCEPT", "DROP", "RETURN"}))
+	if v.DefaultEndpointToHostAction != "" {
+		valid := []string{"ACCEPT", "DROP", "RETURN"}
+		allErrs = append(allErrs, IsValidValue(fldPath, &v.DefaultEndpointToHostAction, valid)...)
 	}
 
-	chainInsertMode := v.ChainInsertMode
-	switch chainInsertMode {
-	case "", "insert", "append":
-	default:
-		allErrs = append(allErrs, field.NotSupported(fldPath.Child("chainInsertMode"), chainInsertMode, []string{"insert", "append"}))
+	if v.ChainInsertMode != "" {
+		valid := []string{"insert", "append"}
+		allErrs = append(allErrs, IsValidValue(fldPath, &v.ChainInsertMode, valid)...)
```

Comment:
Changing from `fldPath.Child("chainInsertMode")` to `fldPath` results in the validation message being less specific, i.e.:
```
# error populating cluster spec: spec.networking.canal: Unsupported value: "bad": supported values: "insert", "append"
```

Can you correct this for all cases here.

Reply:
Thanks for noticing this. I think it should be better now.